### DISCUSSION TITLE
Intuition of the Number of Seasonality Degrees in the Seasonality (Sinusoidal) Function

### DISCUSSION THREAD

**Discussion by saukang** [google/lightweight_mmm#74] [Q&A] [answered]
Hey team,

I was wondering what the seasonality degree represents in the following equation. 
![Screenshot 2022-09-22 145844](https://user-images.githubusercontent.com/71245948/191679526-2da3730f-42d9-4c0e-92a5-182c3393754a.png)
It seems that the seasonality degree is 2 by default. How does it matter with another number of degrees (for instance, 1/3/4/5)?

Also, if I am about to fit monthly-level data, does the equation swap 52 with 12?

Best regards,
Sau Kang.

**Comment by pabloduque0** [comment, marked as answer]
Hello @saukang !

At a high level more degrees accounts for more variation in the seasonality. You can maybe plot a few different outputs to get an idea: 
```
seasonality = media_transforms.calculate_seasonality(your_args_of_choice)
```
Yes, 52 indicates the number of weeks in a year. For daily data we make the adjustments to 365 for you but not for monthly. You can pass that as a transform_kwargs.

**Reply by saukang** [reply]
Thanks @pabloduque0 for your prompt response. Not sure how to pass that as a transform_kwargs but should I be worried with the following setup? 

```
mmm.fit(
    # mandatory inputs
    media=media_data_train,
    media_prior=costs,
    target=target_train,

    # optinal inputs
    extra_features=extra_features_train,
    degrees_seasonality=2,
    seasonality_frequency=12,   # for monthly data
    media_names=df[var_medias].columns,
    number_chains=trial_numbers,

    # custom priors
    custom_priors=custom_priors,

    number_warmup=number_warmup,
    number_samples=number_samples,
    seed=SEED)
```

**Reply by pabloduque0** [reply]
Depends, what model are you using? (That setup with the "carryover" one might have too much lag). For the other two, it should be fine.

Maybe worth also changing the priors of the lagging and/or saturation. I guess most things are setup for either weekly or daily data, we do not envision much monthly data usage.

**Reply by saukang** [reply]
The model name was "hill_adstock" in this setup. Yeah, I thought changing the priors of lagging and saturation for monthly data context would be worth trying too, thanks for the suggestion anyway! I was also thinking about using a combination of carryover + hill transformation for the media variables as well, what do you think about this?

**Reply by pabloduque0** [reply]
At the moment that combination is not provided but we are working on a more modular API that will let you choose many of those things independently. Might still take a month or two until that one is fully released though. In the meantime you might need to fork the code and make some changes if its urgent.

**Reply by saukang** [reply]
Thanks @pabloduque0! Yeah can't wait for that release!
